baseplate/experiments/__init__.py

```diff
@@ -172,54 +194,65 @@ def variant(self, name, user=None, bucketing_event_override=None,
+                now, this expects and will process only the set of
+                user-related fields specified in the user parameter.
+        """
 
-        event = Event("bucketing_events", "bucket")
+        if not self._event_queue or not self._event_constructor:
+            logger.warning("missing event queue or bucketing constructor")
+            return
 
-        context_fields = {}
+        event_context_fields = {}
 
-        if hasattr(self._span.context, "request_context"):
-            request_fields = self._span.context.request_context.event_fields()
-            context_fields.update(request_fields)
+        user_fields = {}
 
         if user:
-            context_fields.update(user.event_fields())
-
-        if "user_logged_in" in context_fields:
-            context_fields["is_logged_out"] = not context_fields.pop("user_logged_in")
-
-        if "cookie_created" in context_fields:
-            context_fields["loid_created"] = context_fields.pop("cookie_created")
+            user_fields['user_id'] = user.event_fields().get("user_id")
+            user_fields['logged_in'] = user.event_fields().get("user_logged_in")
+            user_fields['cookie_created_timestamp'] = user.event_fields().get("cookie_created")
+        else:
+            user_fields['user_id'] = extra_event_fields.get("user_id")
+            user_fields['logged_in'] = extra_event_fields.get("logged_in")
+            user_fields['cookie_created_timestamp'] = extra_event_fields.get("cookie_created_timestamp")
+            user_fields['created_timestamp'] = extra_event_fields.get("created_timestamp")
 
-        for field, value in iteritems(context_fields):
-            event.set_field(field, value)
+        event_context_fields['user'] = user_fields
 
-        for field, value in iteritems(extra_event_fields):
-            event.set_field(field, value)
+        experiment_fields = {}
+        experiment_fields['variant'] = variant
+        experiment_fields['id'] = getattr(experiment, "id", None)
+        experiment_fields['name'] = getattr(experiment, "name", None)
+        experiment_fields['owner'] = getattr(experiment, "owner", None)
+        experiment_fields['version'] = getattr(experiment, "version", None)
 
-        event.set_field("variant", variant)
-        event.set_field("experiment_id", experiment.id)
-        event.set_field("experiment_name", experiment.name)
-        event.set_field("owner", experiment.owner)
+        event_context_fields['experiment'] = experiment_fields
 
-        version = getattr(experiment, "version", None)
-        if version:
-            event.set_field("version", version)
+        event = self._event_constructor(event_context_fields)
 
         span_name = "{}.{}".format(self._context_name, "events")
         with self._span.make_child(span_name) as child_span:
             try:
-                self._event_queue.put(event)
+                serialized = self._event_queue.put(event)
+                print(serialized)
```

Comment:
Looks like you have a leftover print statement

Reply:
gah. Will fix